classes/table/csp_report.php

```diff
@@ -120,12 +144,36 @@ protected function col_action($record) {
         $action = new \confirm_action(get_string('areyousuretodeleteonerecord', 'local_csp'));
         $url = new \moodle_url($this->baseurl);
         $url->params(array(
-            'removerecordwithhash' => $record->sha1hash,
+            'removeviolationclass' => $record->blockeduri,
             'sesskey' => sesskey(),
             'redirecttopage' => $this->currpage,
         ));
         $actionlink = $OUTPUT->action_link($url, get_string('reset', 'local_csp'), $action);
 
         return $actionlink;
     }
+
+    /**
+     * Gets the 3 highest violater documentURIs for each blockedURI
+     *
+     * @param stdObject $record fieldset object of db table with field timecreated
+     * @return string details of the highest violating documents
+     */
+    protected function col_highestviolaters($record) {
+        global $DB, $CFG;
+
+        // Get 3 highest violaters for each blocked URI
+        $sql = "SELECT *
+                  FROM {local_csp}
+                 WHERE blockeduri = ?
+              ORDER BY failcounter DESC";
+        $violaters = $DB->get_records_sql($sql, array($record->blockeduri), 0, 3);
```

Comment:
I'm slightly uneasy about the nest sql, ie 1 sql + 30 nested sql's for each row, if the csp table is big. Fine for now, we need some prod data to see how much of an issue it is or isn't